File: mono-2.6.7-mcs/Skeleton.cs
namespace Skeleton
{
 class Skeleton
 {

  public static void Main()
  {
   TestEnv.TestEnv te= new TestEnv.TestEnv();

   te.CheckComplainAndAdjustExpected( 0);

   te.Test( true);

   te.CheckComplainAndAdjustExpected( 1);

   System.Console.WriteLine( te.Result());

  }
 }
}

File: aspectj-1.2/Skeleton.java

public class Skeleton
{

 public static void main( String[] args)
 {

  TestCode.te.checkComplainAndAdjustExpected( 0);

  TestCode.te.test( 0==TestCode.testCnt);

  TestCode.test();

  TestCode.te.test( 1==TestCode.testCnt);

  TestCode.te.checkComplainAndAdjustExpected( 4);

  System.out.println( ""+ TestCode.te.result());
 }

}

class TestCode
{
 static int testCnt= 0;
 static TestEnv te= new TestEnv();

 public static void test()
 {
  TestCode.te.test( 0==TestCode.testCnt);
  TestCode.testCnt++;
  TestCode.te.test( 1==TestCode.testCnt);
 }
}

aspect TestAspect
{
}
